### PR TITLE
392 retrieve course related ids dynamically

### PR DESCRIPTION
**Pull request by Fiwafoofa**
Resolves: https://github.com/softwareconstruction240/autograder/issues/392

Added support in both frontend and backend for retrieving course related info (assignment IDs, rubric ID/points) from Canvas and setting course related info (course number in addition to what was mentioned before) manually. Removed hardcoded values in PhaseUtils and made methods query the database instead. 

Due to the change in the database, I recommend seeing the modified `README.md` in this pull request. It has a section on updated the old database schema to the new one at the bottom.

### CODE REVIEW

File: src/test/java/integration/CanvasIntegrationImplIT.java [hunk 1]
```diff
@@ -3,43 +3,139 @@
...
-        DaoService.setUserDao(new UserMemoryDao());
-        DaoService.setQueueDao(new QueueMemoryDao());
-        DaoService.setSubmissionDao(new SubmissionMemoryDao());
+    public void setUpEach() throws DataAccessException {
+        DaoService.initializeMemoryDAOs();
+        DaoService.getConfigurationDao().setConfiguration(
+                ConfigurationDao.Configuration.COURSE_NUMBER,
+                26822,
+                Integer.class
+        ); // FIXME: ??? Maybe get dynamically
```
Comment (Fiwafoofa) [inline src/test/java/integration/CanvasIntegrationImplIT.java:42]: Not sure what to do here besides hardcoding the course number. Another potential thing is setting an environment variable like the canvas api token.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,38 @@ property `db.url` is set to `db:3306`
 
 ### Pre-requisites
 
+##### Dev Container (recommended for Windows)
```
Comment (AldenKim) [inline README.md:51]: Having this section for the setup process at the top was very helpful.

**Review by AldenKim — COMMENTED** [review]
All the code to me looks good and ran with no problems on my end.

File: src/test/java/integration/CanvasIntegrationImplIT.java (hunk 1 above)
Comment (cjvuoksi) [reply src/test/java/integration/CanvasIntegrationImplIT.java:42]: I think an environment variable makes sense

**Review by cjvuoksi — COMMENTED** [review]
Naming for updateCanvasRelatedItems bothers me. Could it be more specific?

File: README.md [hunk 3]
```diff
@@ -183,19 +188,120 @@ To generate a Canvas API key:
...
+('Phase4', 'PASSOFF_TESTS', 'Functionality', 'All pass off test cases succeed', 100, '_2614'),
+('Phase4', 'QUALITY', 'Code Quality', 'Chess Code Quality Rubric (see GitHub)', 30, '90346_8398'),
+('Phase4', 'UNIT_TESTS', 'Unit Tests', 'All test cases pass\nEach public method on DAO classes has two test cases, one positive test and one negative test\nEvery test case includes an Assert statement of some type', 25, '90346_5755'),
+('Phase5', 'GIT_COMMITS', 'Git Commits', 'Necessary commit amount', 0, '90347_8497'),
+('Phase5', 'QUALITY', 'Code Quality', 'Chess Code Quality Rubric (see GitHub)', 30, '90347_9378'),
+('Phase5', 'UNIT_TESTS', 'Unit Tests', 'All test cases pass\nEach public method on the Server Facade class has two test cases, one positive test and one negative test\nEvery test case includes an Assert statement of some type', 25, '90347_2215'),
+('Phase6', 'GIT_COMMITS', 'Git Commits', 'Necessary commit amount', 0, '90348_9048'),
+('Phase6', 'PASSOFF_TESTS', 'Automated Pass Off Test Cases', 'There are 20 JUnit pass off test cases. Each successful test case is worth 2.5 points.', 50, '90348_899'),
```
Comment (19mdavenport) [inline README.md:216]: There are no longer 20 cases. I have been using `Each provided test case passed is worth a proportional number of points ((passed / total) * 50)` recently. Does this sound unreasonable? Could you change this to reflect not having 20 cases?

**Review by 19mdavenport — APPROVED** [review]
How much have you tested this? Assuming a reasonable amount of manual testing, this looks good and ready to merge once my comment about the README is addressed (and any of Charles' comments)

**Pull request merged**